src/targets/maven.ts

```diff
@@ -0,0 +1,389 @@
+   * raises an error.
+   *
+   * @returns the maven config for this target.
+   */
+  private getMavenConfig(): MavenTargetConfig {
+    return {
+      ...this.getTargetSecrets(),
+      ...this.getTargetSettings(),
+    };
+  }
+
+  private getTargetSecrets(): Record<TargetSettingType, string> {
+    const secrets = targetSecrets.map(secret => {
+      checkEnvForPrerequisite({ name: secret });
+      return {
+        name: secret,
+        value: process.env[secret],
+      };
+    });
+    return this.reduceConfig(secrets);
+  }
+
+  private reduceConfig(
+    config: { name: string; value: string | undefined }[]
+  ): Record<string, string> {
+    return config.reduce((prev, current) => {
+      return {
+        ...prev,
+        [current.name]: current.value,
+      };
+    }, {});
+  }
+
+  private getTargetSettings(): Record<TargetSettingType, string> {
+    const settings = targetOptions.map(setting => {
+      return {
+        name: setting,
+        value: this.config[setting],
+      };
+    });
+    return this.reduceConfig(settings);
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   * If there's required software missing, raises an error.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug(
+      `Checking if Maven CLI is available on ${this.mavenConfig.mavenCliPath}...`
+    );
+    checkExecutableIsPresent(this.mavenConfig.mavenCliPath);
+    this.logger.debug(
+      `Checking if Gradle CLI is available on ${this.mavenConfig.gradleCliPath}...`
+    );
+    checkExecutableIsPresent(this.mavenConfig.gradleCliPath);
+    this.logger.debug(`Checking if GPG is available in the path...`);
+    checkExecutableIsPresent('gpg');
+  }
+
+  /**
+   * Parses the required target's configuration variables' types from strings to
+   * RegExp.
+   *
+   * To make TS happy, they must be `RegExp` objects. It's not valid to write
+   * `/myRegex/`
+   */
+  private makeRegexpFromConfiguration(): void {
+    this.config.androidDistDirPattern = new RegExp(
+      this.config.androidDistDirPattern
+    );
+    this.config.androidFileSearchPattern = new RegExp(
+      this.config.androidFileSearchPattern
+    );
+  }
+
+  /**
+   * Publishes current Java and Android distributions.
+   * @param version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revison: string): Promise<void> {
+    await this.createUserGradlePropsFile();
+    await this.upload(revison);
+    await this.closeAndRelease();
+  }
+
+  /**
+   * Creates the required user's `gradle.properties` file.
+   *
+   * If there's an existing one, it's overwritten.
+   * TODO: control when it's overwritten with an option.
+   */
+  private async createUserGradlePropsFile(): Promise<void> {
+    await fsPromises.writeFile(
```

Comment:
You can save a cycle by directly returning the `writeFile` promise:
```suggestion
  private async createUserGradlePropsFile(): Promise<number> {
    return fsPromises.writeFile(
```

That said it may affect the stacktrace if an error happens.